sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/DoNotHideBaseClassMethods.cs

```diff
@@ -0,0 +1,178 @@
+                        .WhereNotNull()
+                        .ToList()
+                        .ForEach(c.ReportDiagnostic);
+                },
+                SyntaxKind.ClassDeclaration);
+        }
+
+        private class IssueFinder
+        {
+            private enum Match { Different, Identical, WeaklyDerived }
+
+            private readonly IList<IMethodSymbol> allBaseClassMethods;
+            private readonly SemanticModel semanticModel;
+
+            public IssueFinder(INamedTypeSymbol classSymbol, SemanticModel semanticModel)
+            {
+                this.semanticModel = semanticModel;
+                allBaseClassMethods = GetAllBaseMethods(classSymbol);
+            }
+
+            private static List<IMethodSymbol> GetAllBaseMethods(INamedTypeSymbol classSymbol)
+            {
+                return classSymbol.BaseType
+                         .GetSelfAndBaseTypes()
+                         .SelectMany(t => t.GetMembers())
+                         .OfType<IMethodSymbol>()
+                         .Where(m => IsSymbolVisibleFromNamespace(m, classSymbol.ContainingNamespace))
+                         .Where(m => m.Parameters.Length > 0)
+                         .Where(m => !string.IsNullOrEmpty(m.Name))
+                         .ToList();
+            }
+
+            private static bool IsSymbolVisibleFromNamespace(ISymbol symbol, INamespaceSymbol ns)
+            {
+                return symbol.DeclaredAccessibility != Accessibility.Private &&
+                       (symbol.DeclaredAccessibility != Accessibility.Internal || ns.Equals(symbol.ContainingNamespace));
+            }
+
+            public Diagnostic FindIssue(MemberDeclarationSyntax memberDeclaration)
+            {
+                var methodSymbol = semanticModel.GetDeclaredSymbol(memberDeclaration) as IMethodSymbol;
+                var issueLocation = (memberDeclaration as MethodDeclarationSyntax)?.Identifier.GetLocation();
+
+                if (methodSymbol == null || issueLocation == null)
+                {
+                    return null;
+                }
+
+                var baseMethodHidden = FindBaseMethodHiddenByMethod(methodSymbol);
+                return baseMethodHidden != null ? Diagnostic.Create(rule, issueLocation, baseMethodHidden) : null;
+            }
+
+            private IMethodSymbol FindBaseMethodHiddenByMethod(IMethodSymbol methodSymbol)
+            {
+                var baseMemberCandidates = allBaseClassMethods.Where(m => m.Name == methodSymbol.Name);
+
+                IMethodSymbol hiddenBaseMethodCandidate = null;
+
+                var hasBaseMethodWithSameSignature = baseMemberCandidates.Any(baseMember =>
+                    {
+                        var result = ComputeSignatureMatch(baseMember, methodSymbol);
+                        if (result == Match.WeaklyDerived)
+                        {
+                            hiddenBaseMethodCandidate = hiddenBaseMethodCandidate ?? baseMember;
```

Comment:
No, because there may still be a perfect match with a different method. If there is, the candidate is not returned and there is no issue reported.
Btw, fxcop behaves the same way.